ipfs-cluster-service/main.go

```diff
@@ -261,7 +261,82 @@ removal, upgrade state using this command, and restart every peer.
 					Action: func(c *cli.Context) error {
 						err := upgrade()
 						checkErr("upgrading state", err)
-						return err
+						return nil
+					},
+				},
+				{
+					Name:  "export",
+					Usage: "save the IPFS Cluster state to a human readable json file",
+					Description: `
+
+This command reads the current cluster state and saves it as json for 
+human readability and editing.  In general any version of cluster will only
+work with the latest state format, so out of date snapshots will be migrated
+before the export.
+`,
+					Action: func(c *cli.Context) error {
+						err := export()
```

Comment:
`ipfs-cluster-service state export -f mypath/state.json` should write to mypath/state.json. `ipfs-cluster-service state export` should write to stdout.

Reply:
Is "mypath" the basdir of the cluster config (i.e. same directory as service.json) or cwd?